tezos/api/src/ocaml_conv/from_ocaml.rs

```diff
@@ -0,0 +1,174 @@
+        unsafe {
+            Applied {
+                hash: v.field::<OCamlOperationHash>(0).into_rust(),
+                protocol_data_json: v.field::<OCamlBytes>(1).into_rust(),
+            }
+        }
+    }
+}
+
+unsafe impl FromOCaml<OperationProtocolDataJsonWithErrorListJson>
+    for OperationProtocolDataJsonWithErrorListJson
+{
+    fn from_ocaml(v: OCaml<OperationProtocolDataJsonWithErrorListJson>) -> Self {
+        unsafe {
+            OperationProtocolDataJsonWithErrorListJson {
+                protocol_data_json: v.field::<OCamlBytes>(0).into_rust(),
+                error_json: v.field::<OCamlBytes>(1).into_rust(),
+            }
+        }
+    }
+}
+
+unsafe impl FromOCaml<Errored> for Errored {
+    fn from_ocaml(v: OCaml<Errored>) -> Self {
+        unsafe {
+            Errored {
+                hash: v.field::<OCamlOperationHash>(0).into_rust(),
+                is_endorsement: v.field::<Option<bool>>(1).into_rust(),
+                protocol_data_json_with_error_json: v
+                    .field::<OperationProtocolDataJsonWithErrorListJson>(2)
+                    .into_rust(),
+            }
+        }
+    }
+}
+
+unsafe impl FromOCaml<ValidateOperationResult> for ValidateOperationResult {
+    fn from_ocaml(v: OCaml<ValidateOperationResult>) -> Self {
+        unsafe {
+            ValidateOperationResult {
+                applied: v.field::<OCamlList<Applied>>(0).into_rust(),
+                refused: v.field::<OCamlList<Errored>>(1).into_rust(),
+                branch_refused: v.field::<OCamlList<Errored>>(2).into_rust(),
+                branch_delayed: v.field::<OCamlList<Errored>>(3).into_rust(),
+            }
+        }
+    }
+}
+
+unsafe impl FromOCaml<ValidateOperationResponse> for ValidateOperationResponse {
+    fn from_ocaml(v: OCaml<ValidateOperationResponse>) -> Self {
+        unsafe {
+            ValidateOperationResponse {
+                prevalidator: v.field::<PrevalidatorWrapper>(0).into_rust(),
+                result: v.field::<ValidateOperationResult>(1).into_rust(),
+            }
+        }
+    }
+}
+
+unsafe impl FromOCaml<JsonRpcResponse> for JsonRpcResponse {
+    fn from_ocaml(v: OCaml<JsonRpcResponse>) -> Self {
+        unsafe {
+            JsonRpcResponse {
+                body: v.field::<OCamlBytes>(0).into_rust(),
+            }
+        }
+    }
+}
+
+unsafe impl FromOCaml<Path> for FfiPath {
+    fn from_ocaml(v: OCaml<Path>) -> Self {
+        if v.is_long() {
+            FfiPath(Path::Op)
+        } else {
+            match v.tag_value() {
+                0 => {
+                    let path: FfiPath = unsafe { v.field::<Path>(0).into_rust() };
+                    let right = unsafe { v.field::<OCamlHash>(1).into_rust() };
+
+                    FfiPath(Path::Left(Box::new(PathLeft::new(
+                        path.0,
+                        right,
+                        Default::default(), // TODO: what is body?
```

Comment:
@bkontur do you know that the `body` fields in `PathLeft` and `PathRight` are?

Reply:
@tizoc 
Do you mean BinaryDataCache::default() vs Default::default()?
I see that we use lots of `body: Default::default(),` in tezos/messages module,
so I guess `Default::default()` is ok here

```
#[derive(Clone, Serialize, Deserialize, PartialEq, Debug, Getters)]
pub struct PathLeft {
    #[get = "pub"]
    path: Path,
    #[get = "pub"]
    right: Hash,
    #[serde(skip_serializing)]
    body: BinaryDataCache,
}
```

Reply:
My question is more about what that `body` field is, and if it is something I should care about when performing this mapping. I used `Default::default()` as a placedholder because I needed something in there, but I guess it also works as the correct value to put in there.

Reply:
it is used for kind of cache of encoded/decoded data,
Default::default() is ok, so please, just remove that TODO comment